extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -56,255 +54,391 @@
  */
 public class AzureStorage
 {
-
   // Default value from Azure library
   private static final int DELTA_BACKOFF_MS = 30_000;
 
-  // https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch?tabs=microsoft-entra-id
+  // https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch#request-body
   private static final Integer MAX_MULTI_OBJECT_DELETE_SIZE = 256;
 
-
-  private static final Logger log = new Logger(AzureStorage.class);
+  private static final Logger LOG = new Logger(AzureStorage.class);
 
   private final AzureClientFactory azureClientFactory;
   private final String defaultStorageAccount;
 
   public AzureStorage(
-      AzureClientFactory azureClientFactory,
-      @Nullable String defaultStorageAccount
+      final AzureClientFactory azureClientFactory,
+      final String defaultStorageAccount
   )
   {
     this.azureClientFactory = azureClientFactory;
     this.defaultStorageAccount = defaultStorageAccount;
   }
 
-  public List<String> emptyCloudBlobDirectory(final String containerName, final String virtualDirPath)
+  /**
+   * See {@link AzureStorage#emptyCloudBlobDirectory(String, String, Integer)} for details.
+   */
+  public List<String> emptyCloudBlobDirectory(final String containerName, @Nullable final String prefix)
```

Comment:
i feel like it might be helpful to have two helper functions here (getContainerClient, getOrCreateContainerCilent) and comment on them what they are used for (readonly or r/w) for anyone who's implementing a new api call in this class to figure out which one to use.

alternatively maybe just a comment at the top of the class?

Reply:
Let me add a comment, that's a good idea. I found it easier understanding the code when there was a bit less indirection.